integration-cli/docker_cli_volume_test.go

```diff
@@ -449,3 +452,46 @@ func (s *DockerSuite) TestVolumeCliInspectWithVolumeOpts(c *check.C) {
 	c.Assert(strings.TrimSpace(out), checker.Contains, fmt.Sprintf("%s:%s", k2, v2))
 	c.Assert(strings.TrimSpace(out), checker.Contains, fmt.Sprintf("%s:%s", k3, v3))
 }
+
+func (s *DockerSuite) TestVolumeCliCreateNonIdempotency(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	// Without options
+	name1 := "test1"
+	dockerCmd(c, "volume", "create", "-d", "local", name1)
+	out, _, err := dockerCmdWithError("volume", "create", "-d", "local", name1)
+	c.Assert(err, check.Not(check.IsNil))
+	c.Assert(err.Error(), checker.Contains, fmt.Sprintf("A volume named %s has already been created", name1))
+	c.Assert(out, checker.Contains, fmt.Sprintf("A volume named %s has already been created", name1))
+	// Old remote API will remain idempotent
+	status, b, err := request.SockRequest("POST", "/v1.25/volumes/create", volumetypes.VolumesCreateBody{
+		Driver: "local",
+		Name:   name1,
+	}, daemonHost())
+	c.Assert(err, check.IsNil)
+	c.Assert(status, check.Equals, http.StatusCreated, check.Commentf(string(b)))
+
+	// With options
+	name2 := "test2"
+	dockerCmd(c, "volume", "create", "-d", "local", name2, "--opt", "type=tmpfs", "--opt", "o=size=1m,uid=1000")
+	out, _, err = dockerCmdWithError("volume", "create", "-d", "local", name2, "--opt", "type=tmpfs", "--opt", "o=size=1m,uid=1000")
+	c.Assert(err, check.Not(check.IsNil))
+	c.Assert(err.Error(), checker.Contains, fmt.Sprintf("A volume named %s has already been created", name2))
+	c.Assert(out, checker.Contains, fmt.Sprintf("A volume named %s has already been created", name2))
+	// Old remote API will remain idempotent
+	status, b, err = request.SockRequest("POST", "/v1.25/volumes/create", volumetypes.VolumesCreateBody{
+		Driver: "local",
+		Name:   name2,
+		DriverOpts: map[string]string{
+			"type": "tmpfs",
+			"o":    "size=1m,uid=1000",
+		},
+	}, daemonHost())
+	c.Assert(err, check.IsNil)
+	c.Assert(status, check.Equals, http.StatusCreated, check.Commentf(string(b)))
+
+	out, _, err = dockerCmdWithError("volume", "create", "-d", "local", name2, "--opt", "type=tmpfs", "--opt", "o=size=2m,uid=1000")
```

Comment:
Thanks @AkihiroSuda. The PR has been updated with comments added. The test has also been split. Please take a look.